### ISSUE TITLE
panic with `jj move`

### ISSUE THREAD

**Issue by cdmistman** [martinvonz/jj#3055]
## Steps to Reproduce the Problem

1. create a commit conflict
1. in another working copy do `jj move --from <commit with conflict>`

## Expected Behavior

i should be able to move changes from the conflicted commit into the working copy

## Actual Behavior

```
thread 'main' panicked at 'internal error: entered unreachable code: list of changed files included a merge: Conflicted([Some(File { id: FileId("374d2c56fc96f0b2068ad57bdfafc67b11eceaf7"), executable: true }), Some(File { id: FileId("374d2c56fc96f0b2068ad57bdfafc67b11eceaf7"), executable: false }), None])', cli/src/merge_tools/builtin.rs:167:13
```

## Specifications

- Platform: x64 linux
- Version: `0.14.0-8e4d1af98eccf1ed8a79a7ddf9500467aa7684f5` (installed via nix flake)

**Comment by martinvonz** [comment]
Oh, that looks like a limitation of the new-ish `:builtin` diff editor (looks like you passed `-i` to the command, right?). It should work with e.g. Meld. https://martinvonz.github.io/jj/v0.14.0/config/#editing-diffs has information about configuring a different diff editor.

**Comment by arxanas** [comment]
We should definitely produce a real error message. Or possibly it wouldn't be too hard to simply implement conflicts, since apparently it works for external diff editors already? You wouldn't be able to resolve the conflicts meaningfully in most cases (as the built-in editor is line-oriented), but it should be sufficient to move the lines.

**Comment by martinvonz** [comment]
> We should definitely produce a real error message. Or possibly it wouldn't be too hard to simply implement conflicts, since apparently it works for external diff editors already? You wouldn't be able to resolve the conflicts meaningfully in most cases (as the built-in editor is line-oriented), but it should be sufficient to move the lines.

Yes, it should be pretty easy. We should just materialize them with conflict markers as we do for external tools.

**Comment by arxanas** [comment]
IIRC my main difficulty was determining what new conflicts exist after the move was complete, but presumably it already works/is implemented for external editors.

**Comment by martinvonz** [comment]
For external editors we create one temporary working copy for each side and set the sparse patterns to match only modified files. The the regular snapshotting takes care of parsing the conflict markers. Since the builtin editor doesn't use temporary working copies, we'll have to call the same functions for parsing conflict markers. Here's how we do it in the working copy: https://github.com/martinvonz/jj/blob/89cf6a00a163b40923666010ba5383d6ad67b353/lib/src/local_working_copy.rs#L1146-L1152
